packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -63,24 +63,27 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
     maybeDocuments: MaybeDocument[]
   ): PersistencePromise<void> {
     const promises: Array<PersistencePromise<void>> = [];
-    const documentStore = remoteDocumentsStore(transaction);
 
-    let changedKeys = documentKeySet();
-    for (const maybeDocument of maybeDocuments) {
+    if (maybeDocuments.length > 0) {
+      const documentStore = remoteDocumentsStore(transaction);
```

Comment:
This seems fine...  but why are we calling addEntries() with an empty list anyway?  Should we be filtering this out at a higher-level (potentially preventing _even more_ unnecessary work?

Reply:
I was pondering about doing this one layer up, but ended up picking the `remote_document_cache` for no particular reason (other than this would solve the same problem no matter who calls `addEntries`). Leaving as is.